### PR TITLE
doc: add pre/postconditions and invarians to module keymeta

### PR DESCRIPTION
**Pull request by lawli3t**
## Basics

These points need to be fulfilled for every PR:

- [ ] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

File: src/libs/elektra/keymeta.c [hunk 1]
```diff
@@ -488,6 +493,10 @@ const Key * keyGetMeta (const Key * key, const char * metaName)
  * If @p metaName does not start with 'meta:/',
  * it will be prefixed with 'meta:/'.
  *
+ * @pre @p metaName is prefixed with "meta:/"
+ * @pre @p key's metadata is not read-only
+ * @post keyGetMeta(key, metaName) == newMetaString
```
Comment (markus2330) [inline src/libs/elektra/keymeta.c:498]: better to make a sentence if it does not necessarily compile&work as written

File: src/libs/elektra/keymeta.c [hunk 2]
```diff
@@ -609,6 +618,11 @@ ssize_t keySetMeta (Key * key, const char * metaName, const char * newMetaString
  * @note You must not delete the returned KeySet.
  * @note Adding a key with metadata to the KeySet is an error.
  *
+ * @post for the returned KeySet ks: keyGetMeta(key, metaName) ==
+ * ksLookupByName(ks, metaName)
+ * @post if @p key did not contain metadata, a new KeySet for the metadata will
```
Comment (markus2330) [inline src/libs/elektra/keymeta.c:623]: Simply write that there will be a KeySet for the metadata, it does not matter if and when it gets created

**Review by markus2330 — COMMENTED** [review]
Very good work! My comments are only about minor nuances.

**Comment by mpranj** [comment]
Thank you!

**Pull request merged**